test/Test/TomlSpec.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Test.TomlSpec where
+
+import Relude (Last (Last), Maybe (Just), Text, pure, ($), (.), (<$>), (<*>))
+
+import Hedgehog (MonadGen, forAll, property, tripping)
+import Test.Tasty (TestTree)
+import Test.Tasty.Hedgehog (testProperty)
+import Toml.Bi.Code (decode, encode)
+
+import Summoner.Config (ConfigP (..), PartialConfig, configT)
+import Summoner.License (License (..), licenseNames)
+import Summoner.ProjectData (CustomPrelude (..), Decision, GhcVer (..))
+
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+genDecision :: MonadGen m => m Decision
+genDecision = Gen.enumBounded
+
+genText :: MonadGen m => m Text
+genText = Gen.text
+    (Range.constant 0 10)
+    (Gen.frequency [ (1, Gen.element ['.','-']), (10, Gen.alphaNum) ])
+
+genTextArr :: MonadGen m => m [Text]
+genTextArr = Gen.list (Range.constant 0 10) genText
+
+genGhcVerArr :: MonadGen m => m [GhcVer]
+genGhcVerArr = Gen.list (Range.constant 0 10) Gen.enumBounded
+
+genCustomPrelude :: MonadGen m => m CustomPrelude
+genCustomPrelude = Prelude
+    <$> genText
+    <*> genText
```

Comment:
I think it's okay to write this function in single line. Usually splitting to multiple lines make sense when the line becomes too big.